app/Services/ComposeDiff.php

```
@@ -0,0 +1,225 @@
+     * @return array<int, array{index: int, lines: array<int, array{type: string, text: string}>}>
+     */
+    public static function hunks(string $current, string $latest): array
+    {
+        $old = self::lines($current);
+        $new = self::lines($latest);
+        $groups = self::groupOpcodes(self::opcodes($old, $new));
+
+        $hunks = [];
+        foreach ($groups as $i => $group) {
+            $lines = [];
+            foreach ($group as [$tag, $i1, $i2, $j1, $j2]) {
```

PHPMD (2.15.0)

[warning] Avoid variables with short names like $i1. Configured minimum length is 3. (undefined)

(ShortVariable)